akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1933,6 +1933,22 @@ trait FlowOpsMat[+Out, +Mat] extends FlowOps[Out, Mat] {
     viaMat(GraphStages.terminationWatcher)(matF)
 
   /**
+    * Materializes to `FlowMonitor[Out]` that allows monitoring of the the current flow. All events are propagated
+    * by the monitor unchanged. Note that the monitor inserts a memory barrier every time it processes an
+    * event, and may therefor affect performance.
+    */
+  def monitor: ReprMat[Out, FlowMonitor[Out]] = monitorMat()(Keep.right)
```

Comment:
@drewhk If we ignore the materialized value of the flow, then there's no point in adding the monitor. Maybe I should just leave the 'mat' version, like watchTermination?